### ISSUE TITLE
(Suggestion) MAC IP adress ban

### ISSUE THREAD

**Issue by MithrandirCraft** [grandwazir/BanHammer#120]
Hello. I would like to suggest an extra expulsion feature for this plugin.

Sometimes it just isn't enough whith an IP ban in order to block bad players. Some routers reset their network IP adress just whith a simple reset.
Proxies and VPN's are also valid ways around the issue of getting banned from a server.

In fact, I've already suggested this in the past, but if it where possible to ban MAC IP adresses whith BanHammer, anyone could effectively block certain systems from entering the server. And MAC adresses are something people don't often modify or reset. So BanHammer would be providing a very effective tool against players who don't like playing by the rules.

In conclusion, Ban Hammer could benefit from some extra access denial features, and I personally suggest a MAC adress expulsion feature.

**Comment by QueueAmI** [comment]
MAC address banning won't work across the Internet. Every router replaces
the MAC with its own as it forwards it. You can only get the MAC of the
last router it passed through, most likely the one your computer is
connected to.

**Comment by MithrandirCraft** [comment]
@QueueAmI I never got to answer this, but I'm wondering about what effective ways I have in order to permanently ban users from joining with a new account.

**Comment by QueueAmI** [comment]
Right now, an IP ban is really your only option. We have to deal with this
too and it's really annoying.

That said, I wonder if there's some client/computer information that could
be grabbed from the client when they log in. Kind of the way Internet
browsers work. If so, there might be something in there that could be used
for a more permanent ban.

**Comment by grandwazir** [comment]
More importantly is that at the level that Java works you are not able to get at a client's MAC easily anyway. 

To resolve this I did create an addon plugin to BanHammer called Alias. What this plugin did was record the names and IP addresses of players who logged in. It then used this information to automatically ban someone who logged in using a different account but from the same IP address. The ban it would issue would be the same as the original ban. It isn't perfect but does make it more difficult for someone to circumvent bans. If they change their IP address and try to login you would need to ban them again manually for the system to make the link. That might be what you are looking for.

**Comment by MithrandirCraft** [comment]
@grandwazir I don't really understand what your addon does.

For what I interpret with what you said, "Alias" blocks all players that login from an IP, once you have banned one of them... I could just search what other accounts are registred with the IP, and ban the ones I chose. It's not very usefull for when the player logs from another IP...

Yeah, it doesn't look perfect u_u

**Comment by grandwazir** [comment]
@MithrandirCraft You could do it manually as well. The main aim was to stop casual attackers. If someone is determined enough to change addresses after each time they are banned it would not help.

**Comment by MithrandirCraft** [comment]
@grandwazir Thank god what players hate most is losing their progress. :P

Just consider any solution you may find or think of :)
